src/expr/src/scalar/func.rs

```diff
@@ -1631,101 +1632,58 @@ fn range_difference<'a>(
     l.difference(&r)?.into_result(temp_storage)
 }
 
-#[sqlfunc(
-    output_type = "bool",
-    is_infix_op = true,
-    sqlname = "=",
-    propagates_nulls = true,
-    negate = "Some(NotEq.into())"
-)]
-fn eq<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+#[sqlfunc(is_infix_op = true, sqlname = "=", negate = "Some(NotEq.into())")]
+fn eq<'a>(a: ExcludeNull<Datum<'a>>, b: ExcludeNull<Datum<'a>>) -> bool {
     // SQL equality demands that if either input is null, then the result should be null. However,
     // we don't need to handle this case here; it is handled when `BinaryFunc::eval` checks
     // `propagates_nulls`.
-    if a == Datum::Null || b == Datum::Null {
-        Datum::Null
-    } else {
-        Datum::from(a == b)
-    }
+    a == b
```

Comment:
It's great that this comment got true again. It used to be true, and then not true, and now true again. :)